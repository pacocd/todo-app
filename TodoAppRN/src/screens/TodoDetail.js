import React, { Component } from 'react';
import { View, Text, Image, ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import { images } from '../assets';

export class TodoDetail extends Component {
  static propTypes = {
    navigation: PropTypes.shape({
      state: PropTypes.shape({
        params: PropTypes.shape({
          todo: PropTypes.shape({
            id: PropTypes.number.isRequired,
            user_id: PropTypes.number.isRequired,
            title: PropTypes.string.isRequired,
            completed: PropTypes.bool.isRequired
          }).isRequired
        }).isRequired
      }).isRequired
    }).isRequired
  };

  static navigationOptions = {
    title: 'Detail'
  };

  renderLabels(title, content) {
    return (
      <View style={{ margin: 15 }}>
        <Text style={{ fontWeight: 'bold', marginBottom: 8, fontSize: 12 }}>
          {title}
        </Text>
        <Text style={{ fontSize: 14 }}>{content}</Text>
      </View>
    );
  }

  renderCompletionStatusImage(completed) {
    const image = completed ? images.checkmark : images.xIcon;

    return (
      <Image
        resizeMode="contain"
        source={image}
        style={{
          alignSelf: 'center',
          height: 100,
          width: 100,
          marginVertical: 20
        }}
      />
    );
  }

  render() {
    const {
      id,
      user_id: userId,
      title,
      completed
    } = this.props.navigation.state.params.todo;
    return (
      <ScrollView style={{ flex: 1, backgroundColor: 'white' }}>
        {this.renderCompletionStatusImage(completed)}
        {this.renderLabels('Id', id)}
        {this.renderLabels('User', userId)}
        {this.renderLabels('Title', title)}
      </ScrollView>
    );
  }
}

export default TodoDetail;
