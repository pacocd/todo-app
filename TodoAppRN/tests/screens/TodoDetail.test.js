import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoDetail from '../../src/screens/TodoDetail';

configure({ adapter: new Adapter() });

describe('when TodoDetail is fully rendered', () => {
  const props = {
    navigation: {
      state: {
        params: { todo: { id: 1, title: 'test', user_id: 1, completed: false } }
      }
    }
  };
  const wrapper = shallow(<TodoDetail {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
