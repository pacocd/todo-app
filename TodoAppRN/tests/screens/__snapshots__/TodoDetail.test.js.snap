// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when TodoDetail is fully rendered renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoDetail
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "todo": Object {
              "completed": false,
              "id": 1,
              "title": "test",
              "user_id": 1,
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/images/x_icon.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 100,
              "marginVertical": 20,
              "width": 100,
            }
          }
        />,
        <Component
          style={
            Object {
              "margin": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              }
            }
          >
            Id
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            1
          </Text>
        </Component>,
        <Component
          style={
            Object {
              "margin": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              }
            }
          >
            User
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            1
          </Text>
        </Component>,
        <Component
          style={
            Object {
              "margin": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              }
            }
          >
            Title
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            test
          </Text>
        </Component>,
        <SquareButton
          color="#4688f1"
          onPress={[Function]}
          text="Edit"
          textColor="white"
        />,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resizeMode": "contain",
          "source": Object {
            "testUri": "../../../src/assets/images/x_icon.png",
          },
          "style": Object {
            "alignSelf": "center",
            "height": 100,
            "marginVertical": 20,
            "width": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              Id
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              1
            </Text>,
          ],
          "style": Object {
            "margin": 15,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Id",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              },
            },
            "ref": null,
            "rendered": "Id",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": 1,
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              User
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              1
            </Text>,
          ],
          "style": Object {
            "margin": 15,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "User",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              },
            },
            "ref": null,
            "rendered": "User",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": 1,
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              Title
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              test
            </Text>,
          ],
          "style": Object {
            "margin": 15,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Title",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 8,
              },
            },
            "ref": null,
            "rendered": "Title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "test",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#4688f1",
          "onPress": [Function],
          "text": "Edit",
          "textColor": "white",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/x_icon.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "marginVertical": 20,
                "width": 100,
              }
            }
          />,
          <Component
            style={
              Object {
                "margin": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              Id
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              1
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "margin": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              User
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              1
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "margin": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                }
              }
            >
              Title
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              test
            </Text>
          </Component>,
          <SquareButton
            color="#4688f1"
            onPress={[Function]}
            text="Edit"
            textColor="white"
          />,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": Object {
              "testUri": "../../../src/assets/images/x_icon.png",
            },
            "style": Object {
              "alignSelf": "center",
              "height": 100,
              "marginVertical": 20,
              "width": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                  }
                }
              >
                Id
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                1
              </Text>,
            ],
            "style": Object {
              "margin": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Id",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                },
              },
              "ref": null,
              "rendered": "Id",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": 1,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                },
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                  }
                }
              >
                User
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                1
              </Text>,
            ],
            "style": Object {
              "margin": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "User",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                },
              },
              "ref": null,
              "rendered": "User",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": 1,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                },
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                  }
                }
              >
                Title
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                test
              </Text>,
            ],
            "style": Object {
              "margin": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Title",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 8,
                },
              },
              "ref": null,
              "rendered": "Title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "test",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                },
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#4688f1",
            "onPress": [Function],
            "text": "Edit",
            "textColor": "white",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
